src/hooks/usePageReducer.js

```diff
@@ -40,24 +41,34 @@ const usePageReducer = (
   const columns = useMemo(() => getColumns(page), [page]);
   const pageInfo = useMemo(() => getPageInfo(page), [page]);
   const memoizedReducer = useMemo(() => getReducer(page), []);
-  const [state, dispatch] = useReducer(memoizedReducer, {
+
+  const [pageState, setPageState] = useState({
     ...initialState,
     columns,
     pageInfo,
   });
 
-  const thunkDispatcher = action => {
-    if (typeof action === 'function') action(dispatch, state);
-    else dispatch(action);
+  // Reference to the current state object, indepdent of closures.
+  const stateRef = useRef(pageState);
+  const getState = () => stateRef.current;
+
+  // Basic dispatch function.
+  const dispatch = action => {
+    const newState = memoizedReducer(getState(), action);
+    setPageState(newState);
+    stateRef.current = newState;
   };
```

Comment:
If we do figure a better way to do this, then it's easy enough to refactor this anyway.